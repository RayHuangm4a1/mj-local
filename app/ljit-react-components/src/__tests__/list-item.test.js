import React from 'react';
import { shallow, mount, } from 'enzyme';

describe('ListItem', () => {
	let ListItem;

	beforeEach(() => {
		jest.mock('antd/lib/list');
		ListItem = require('../components/list-item').default;
	});

	afterEach(() => {
		jest.resetModules();
	});

	it('should render correctly', () => {
		const className = 'mock-class';
		const title = 'mock-title';
		const titleHint = 'mock-title-hint';
		const content = 'mock-content';
		const right = 'mock-right';
		const prefix = 'mock-prefix';
		const wrapper = shallow(
			<ListItem
				className={className}
				title={title}
				titleHint={titleHint}
				content={content}
				right={right}
				prefix={prefix}
			/>
		);

		expect(wrapper).toMatchSnapshot();
	});

	it('should be selectable by class ljit-list-item', () => {
		const title = 'mock-title';
		const titleHint = 'mock-title-hint';
		const content = 'mock-content';
		const right = 'mock-right';
		const wrapper = shallow(
			<ListItem
				title={title}
				titleHint={titleHint}
				content={content}
				right={right}
			/>
		);

		expect(wrapper.hasClass('ljit-list-item')).toEqual(true);
	});

	it('should be selectable by custom class', () => {
		const className = 'mock-class';
		const wrapper = shallow(<ListItem className={className} />);

		expect(wrapper.hasClass(className)).toEqual(true);
	});

	it('should mount in a full DOM', () => {
		const className = 'mock-class';
		const title = 'mock-title';
		const titleHint = 'mock-title-hint';
		const content = 'mock-content';
		const right = 'mock-right';
		const prefix = 'mock-prefix';
		const wrapper = mount(
			<ListItem
				className={className}
				title={title}
				titleHint={titleHint}
				content={content}
				right={right}
				prefix={prefix}
			/>
		);

		expect(wrapper.props().className).toBe(className);
		expect(wrapper.props().title).toBe(title);
		expect(wrapper.props().titleHint).toBe(titleHint);
		expect(wrapper.props().content).toBe(content);
		expect(wrapper.props().right).toBe(right);
	});
});
