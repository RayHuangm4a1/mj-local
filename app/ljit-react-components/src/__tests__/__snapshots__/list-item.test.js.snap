// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListItem
    className="mock-class"
    content="mock-content"
    prefix="mock-prefix"
    right="mock-right"
    title="mock-title"
    titleHint="mock-title-hint"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="ljit-list-item__wrapper"
      >
        mock-prefix
        <div>
          <div
            className="ljit-list-item__title"
          >
            mock-title
            <span
              className="ljit-list-item__title-hint"
            >
              mock-title-hint
            </span>
          </div>
          mock-content
        </div>
      </div>,
      "className": "ljit-list-item mock-class",
      "extra": "mock-right",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "mock-prefix",
          <div>
            <div
              className="ljit-list-item__title"
            >
              mock-title
              <span
                className="ljit-list-item__title-hint"
              >
                mock-title-hint
              </span>
            </div>
            mock-content
          </div>,
        ],
        "className": "ljit-list-item__wrapper",
      },
      "ref": null,
      "rendered": Array [
        "mock-prefix",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="ljit-list-item__title"
              >
                mock-title
                <span
                  className="ljit-list-item__title-hint"
                >
                  mock-title-hint
                </span>
              </div>,
              null,
              "mock-content",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "mock-title",
                  <span
                    className="ljit-list-item__title-hint"
                  >
                    mock-title-hint
                  </span>,
                ],
                "className": "ljit-list-item__title",
              },
              "ref": null,
              "rendered": Array [
                "mock-title",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "mock-title-hint",
                    "className": "ljit-list-item__title-hint",
                  },
                  "ref": null,
                  "rendered": "mock-title-hint",
                  "type": "span",
                },
              ],
              "type": "div",
            },
            null,
            "mock-content",
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [MockFunction],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="ljit-list-item__wrapper"
        >
          mock-prefix
          <div>
            <div
              className="ljit-list-item__title"
            >
              mock-title
              <span
                className="ljit-list-item__title-hint"
              >
                mock-title-hint
              </span>
            </div>
            mock-content
          </div>
        </div>,
        "className": "ljit-list-item mock-class",
        "extra": "mock-right",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "mock-prefix",
            <div>
              <div
                className="ljit-list-item__title"
              >
                mock-title
                <span
                  className="ljit-list-item__title-hint"
                >
                  mock-title-hint
                </span>
              </div>
              mock-content
            </div>,
          ],
          "className": "ljit-list-item__wrapper",
        },
        "ref": null,
        "rendered": Array [
          "mock-prefix",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="ljit-list-item__title"
                >
                  mock-title
                  <span
                    className="ljit-list-item__title-hint"
                  >
                    mock-title-hint
                  </span>
                </div>,
                null,
                "mock-content",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "mock-title",
                    <span
                      className="ljit-list-item__title-hint"
                    >
                      mock-title-hint
                    </span>,
                  ],
                  "className": "ljit-list-item__title",
                },
                "ref": null,
                "rendered": Array [
                  "mock-title",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "mock-title-hint",
                      "className": "ljit-list-item__title-hint",
                    },
                    "ref": null,
                    "rendered": "mock-title-hint",
                    "type": "span",
                  },
                ],
                "type": "div",
              },
              null,
              "mock-content",
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [MockFunction],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
